### PR TITLE
Fix crashing syncing thread when logging big integers

### PR DESCRIPTION
**Pull request by aniezurawski**

### CODE REVIEW

File: src/neptune/attributes/atoms/integer.py [hunk 1]
```diff
@@ -45,5 +53,17 @@ def assign(self, value: typing.Union[IntegerVal, float, int], *, wait: bool = Fa
         if not isinstance(value, IntegerVal):
             value = IntegerVal(value)
 
+        if value.value > Integer.MAX_32_BIT_INT or value.value < Integer.MIN_32_BIT_INT:
```
Comment (AleksanderWWW) [inline src/neptune/attributes/atoms/integer.py:56]: Slightly more readable syntax:
`Integer.MIN_32_BIT_INT < value.value < Integer.MAX_32_BIT_INT`

**Review by AleksanderWWW — APPROVED** [review]
LGTM. One optional thing to change

File: src/neptune/attributes/atoms/integer.py [hunk 2]
```diff
@@ -45,5 +50,17 @@ def assign(self, value: typing.Union[IntegerVal, float, int], *, wait: bool = Fa
...
+            logger.warning(
+                "WARNING: The value you're trying to log is outside the range of 32-bit integers "
+                "(%s to %s) and will be skipped. "
+                "We'll support 64-bit integers in the future. "
+                'For now, try logging the value as a float instead: run["field"] = float(%s)',
+                Integer.MIN_32_BIT_INT,
+                Integer.MAX_32_BIT_INT,
+                value.value,
+            )
+            return
```
Comment (Raalsky) [inline src/neptune/attributes/atoms/integer.py:63]: `return None` is more convenient.

**Review by Raalsky — CHANGES_REQUESTED** [review]

File: src/neptune/attributes/atoms/integer.py (hunk 2 above)
Comment (aniezurawski) [reply src/neptune/attributes/atoms/integer.py:63]: Is it, really? For function which is meant to be a procedure? Seems misleading.

Comment (Raalsky) [reply src/neptune/attributes/atoms/integer.py:63]: Ok. ChatGPT agrees with you :P.

**Review by Raalsky — APPROVED** [review]

**Review by Raalsky — APPROVED** [review]

**Pull request merged**